### PR TITLE
Add dashboard crds using manifests

### LINKED ISSUE #531: Move Dashboard CRDs to odh-dashboard repo
ODH Dashboard CRDs are bundled in the operator [bundle](https://github.com/opendatahub-io/opendatahub-operator/tree/main/bundle/manifests). Use `crd` folder from [odh-dashboard](https://github.com/opendatahub-io/odh-dashboard/tree/main/manifests/crd) to deploy required CRDs.

This will allow following features:
- Installing dashboard CRDs on when managementState is `Managed` for odh-dashboard
- Installing CRDs depending on the dashboard branch.

### PR DESCRIPTION
**Pull request by VaishnaviHire**
Fixes #531 
## Description
This PR  introduces following changes:
- Deploy Dashboard CRDs only when odh-dashboard component is enabled
- Remove Dashboard CRDs from bundle
- Remove CRDs from validation test in e2e framework

## How Has This Been Tested?
1. Deploy test catalogsource
```
quay.io/vhire/opendatahub-operator-catalog:v2.1.10
```

2. Verify Dashboard CRDs are not installed as part of bundle
Screenshot 2023-09-13 at 3 49 33 PM

3. Verify CRDs are deployed when dasboard is enabled
## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/dashboard/dashboard.go [hunk 1]
```diff
@@ -68,12 +69,31 @@ func (d *Dashboard) ReconcileComponent(cli client.Client, owner metav1.Object, d
 			if err != nil {
 				return err
 			}
+
+			// Deploy CRDs for odh-dashboard
```
Comment (zdtsw) [inline components/dashboard/dashboard.go:73]: should this block put out of the if-block? 
if we do not care which platform as long as it is enabled component

**Comment by zdtsw** [comment]
also thinking, do we need some state this change in the release? since the screenshot for installation was done with all these 4 CRDs in

File: components/dashboard/dashboard.go (hunk 1 above)
Comment (zdtsw) [reply components/dashboard/dashboard.go:73]: another thing, in deploy.go https://github.com/opendatahub-io/opendatahub-operator/blob/f43f09b30be92fbc8984dce20a00a7fe79e8f3ce/pkg/deploy/deploy.go#L233 should we update to 
```suggestion
if found.GetKind() == "CustomResourceDefinition" && compFound != "odh-dashboard" {
			// Deploy CRDs for odh-dashboard
				return nil
			}
```

Comment (VaishnaviHire) [reply components/dashboard/dashboard.go:73]: Also for the deploy, I think we should not be cleaning up the CRDs even for dashboard...right ?

Comment (zdtsw) [reply components/dashboard/dashboard.go:73]: i am not really sure about the CRD  part. 
if UI team is keeping update CRD schema, when an old CRD has been left in the cluster, will the new one be applied to cluster (assume both with same api version v1alpha1) unless they are always backwards compatible, e.g no need conversion.

Comment (zdtsw) [reply components/dashboard/dashboard.go:73]: @etirelli how you feel about these dashboard CRDs, should remove them if dashboard is disabled or just keep them?

Comment (zdtsw) [reply components/dashboard/dashboard.go:73]: ok, we can keep CRD in the cluster, as long as UI team make sure any new changes are backwards compatible e.g when add the new CRD to overwrite the old CRD wont be any conflict

File: tests/e2e/odh_manager_test.go [hunk 2]
```diff
@@ -27,17 +27,4 @@ func (tc *testContext) validateOwnedCRDs(t *testing.T) {
 
 	require.NoErrorf(t, tc.validateCRD("dscinitializations.dscinitialization.opendatahub.io"),
 		"error in validating CRD : dscinitializations.dscinitialization.opendatahub.io")
-
```
Comment (zdtsw) [inline tests/e2e/odh_manager_test.go:30]: can we move these 4 into https://github.com/opendatahub-io/opendatahub-operator/blob/main/tests/e2e/dsc_creation_test.go#L340

File: components/dashboard/dashboard.go [hunk 3]
```diff
@@ -68,12 +69,31 @@ func (d *Dashboard) ReconcileComponent(cli client.Client, owner metav1.Object, d
...
 			err := common.UpdatePodSecurityRolebinding(cli, []string{"rhods-dashboard"}, dscispec.ApplicationsNamespace)
 			if err != nil {
 				return err
 			}
+
+			// Deploy CRDs for odh-dashboard
+			err = deploy.DeployManifestsFromPath(owner, cli, ComponentNameSupported,
+				PathCRDs,
+				dscispec.ApplicationsNamespace,
+				cli.Scheme(), enabled)
```
Comment (zdtsw) [inline components/dashboard/dashboard.go:93]:
```suggestion
			err = deploy.DeployManifestsFromPath(cli, owner,
				PathCRDs,
				dscispec.ApplicationsNamespace,
				ComponentNameSupported,
				enabled)
```

File: components/dashboard/dashboard.go [hunk 4]
```diff
@@ -68,12 +69,31 @@ func (d *Dashboard) ReconcileComponent(cli client.Client, owner metav1.Object, d
 			if err != nil {
 				return err
 			}
+
+			// Deploy CRDs for odh-dashboard
+			err = deploy.DeployManifestsFromPath(owner, cli, ComponentName,
+				PathCRDs,
+				dscispec.ApplicationsNamespace,
+				cli.Scheme(), enabled)
```
Comment (zdtsw) [inline components/dashboard/dashboard.go:77]:
```suggestion
			err = deploy.DeployManifestsFromPath(cli, owner,,
				PathCRDs,
				dscispec.ApplicationsNamespace,
				ComponentName,
				enabled)
```

**Review by zdtsw — COMMENTED** [review]
function parameters changed after rebase

File: components/dashboard/dashboard.go [hunk 5]
```diff
@@ -68,12 +69,33 @@ func (d *Dashboard) ReconcileComponent(cli client.Client, owner metav1.Object, d
 			if err != nil {
 				return err
 			}
+
+			// Deploy CRDs for odh-dashboard
+			err = deploy.DeployManifestsFromPath(cli, owner,,
```
Comment (zdtsw) [inline components/dashboard/dashboard.go:74]:
```suggestion
			err = deploy.DeployManifestsFromPath(cli, owner,
```

**Pull request merged**